### ISSUE TITLE
Mysql CPU limit value not set

### ISSUE THREAD

**Issue by kevprice83** [3scale/3scale-amp-openshift-templates#5]
If a project has default quotas set then these values will be respected if they are not overridden by the template value. That can be an issue when only the request value has been declared in the template and the limit in the project is less than the request value.

I ran into this issue when deploying the AMP template into a project that had a default CPU limit of 800mi. The request value can be seen [here](https://github.com/3scale/3scale-amp-openshift-templates/blob/master/amp/amp.yml#L1067-L1071) being set at 1 core. The limit value hasn't been set and probably should to avoid this kind of issue. This was fixed by adding the line `cpu: 2` after line 1067 in the limits.

**Comment by andrewdavidmackenzie** [comment]
Not sure it's clear what should be in the template here... will depend on cores in the machine and other factors, no?

**Comment by mikz** [comment]
We don't really plan to limit CPU of MySQL. Why should we do that? 

So the issue here is that the limit was 800mi and the request 1000mi ? OpenShift was rejecting it or what was the actual issue.

**Comment by kevprice83** [comment]
@mikz yes we are setting a request value so if the project has a default limit which is less than that of the request value in the template then it will fail on deployment.

**Comment by kevprice83** [comment]
So from an OpenShift perspective it is recommended to set either both the `limit` and the `request` or neither. I understand that you don't want to "limit" the CPU on the MySQL but the reality is there will not be infinite resources and once the request value is set this will be honoured every time until the project limit has been reached and then it will run into issues.

**Comment by mikz** [comment]
@kevprice83 OpenShift will reject that template or what will happen? Is there some error message ?

**Comment by kevprice83** [comment]
The template is valid as far as the oc binary is concerned but there will then be an error message in the MySQL pod regarding limits so it just won't deploy. Once there is a bit of troubleshooting done then sure you can find the culprit and change the template, see [docs](https://docs.openshift.com/container-platform/3.4/dev_guide/compute_resources.html#dev-requests-vs-limits) for more details on how OpenShift enforces this.

**Comment by kevprice83** [comment]
@mikz should I move this issue to the open shift repo we have?

**Comment by mikz** [comment]
The upstream template is here so would be good to fix it here too.
